src/app/GitUI/Editor/ComboBoxWatermarkManager.cs

```diff
@@ -0,0 +1,185 @@
+namespace GitUI.Editor;
+
+/// <summary>
+///  Manages watermark text display for a ComboBox control.
+/// </summary>
+internal sealed class ComboBoxWatermarkManager : IDisposable
+{
+    private readonly ComboBox _comboBox;
+    private readonly string _watermarkText;
+    private Font _watermarkFont;
+    private readonly Color _watermarkColor = SystemColors.GrayText;
+    private Font _originalFont;
+    private Color _originalForeColor;
+    private bool _isWatermarkActive;
+    private bool _isDisposed;
+
+    /// <param name="comboBox">The ComboBox control to manage.</param>
+    /// <param name="watermarkText">The watermark text to display.</param>
+    public ComboBoxWatermarkManager(ComboBox comboBox, string watermarkText)
```

Comment:
[docs] Please add complete set of docs (i.e., summary, params, return, etc.) for all public members.